scripts/check-eslint-windows.test.sh

```
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+# Self-test for the eslint.config.js Windows-portability rules (#1623).
+#
+# `npm run lint:windows` only proves the config RUNS clean on the (clean) tree —
+# a silently-broken selector would pass that too. This confirms each rule still
+# FIRES on its anti-pattern and stays quiet on the fixed form, using a temp
+# fixture under the config's `files` glob (cleaned up on exit).
+set -uo pipefail
+ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
+cd "$ROOT"
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)